realm/src/androidTest/java/io/realm/RealmObjectSchemaTests.java

```diff
@@ -0,0 +1,641 @@
+                    schema.addInt(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case LONG:
+                    fieldName = AllJavaTypes.FIELD_LONG;
+                    schema.addLong(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case BYTE:
+                    fieldName = AllJavaTypes.FIELD_BYTE;
+                    schema.addByte(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case BOOLEAN:
+                    fieldName = AllJavaTypes.FIELD_BOOLEAN;
+                    schema.addBoolean(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case FLOAT:
+                    fieldName = AllJavaTypes.FIELD_FLOAT;
+                    schema.addFloat(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case DOUBLE:
+                    fieldName = AllJavaTypes.FIELD_DOUBLE;
+                    schema.addDouble(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case BLOB:
+                    fieldName = AllJavaTypes.FIELD_BLOB;
+                    schema.addBlob(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case DATE:
+                    fieldName = AllJavaTypes.FIELD_DATE;
+                    schema.addDate(fieldName);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case OBJECT:
+                    fieldName = AllJavaTypes.FIELD_OBJECT;
+                    schema.addObject(fieldName, DOG_SCHEMA);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                case LIST:
+                    fieldName = AllJavaTypes.FIELD_LIST;
+                    schema.addObject(fieldName, DOG_SCHEMA);
+                    checkAddedAndRemovable(fieldName);
+                    break;
+                default:
+                    fail();
+            }
+        }
+    }
+
+    // Check that field is actually added and that i can be removed again.
+    private void checkAddedAndRemovable(String fieldName) {
+        assertTrue(schema.hasField(fieldName));
+        schema.removeField(fieldName);
+        assertFalse(schema.hasField(fieldName));
+    }
+
+    public void testAddRemoveIndexedField() {
+        for (IndexFieldType fieldType : IndexFieldType.values()) {
+            String fieldName;
+            switch (fieldType) {
+                case STRING:
+                    fieldName = AllJavaTypes.FIELD_STRING;
+                    schema.addString(fieldName, EnumSet.of(RealmModifier.INDEXED));
+                    checkAddedAndRemovable(fieldName);
+                    break;
+//                case SHORT:
```

Comment:
Not really, but it will most likely be merged in just before this one. So call it prematurely optimisation while I was touching the code anyway :). It is the same with Null, these features kinda affect all ongoing work that depend on knowing what types are available in Realm.